### PR TITLE
Driver module parameters for Mellanox based adapters.

### PR DESCRIPTION
**Pull request by manvanthar**
Driver module parameters for Mellanox based adapters.

Signed-off-by: Manvanthara Puttashankar <manvanth@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by manvanthar** [comment]
avocado run driver_parameter.py -m driver_parameter.py.data/driver_parameter_mlx5_core.yaml 
Failed to load plugin from module "avocado_runner_vm": DistributionNotFound(Requirement.parse('libvirt-python'), None) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-72.0-py3.6.egg/avocado/core/extension_manager.py", line 62, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2317, in load
    self.require(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2340, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 774, in resolve
    raise DistributionNotFound(req, requirers)

JOB ID     : ac3b9f47eb54525b09edff3f4314eb7fa45cb766
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-12-03T20.51-ac3b9f4/job.log
 (01/12) driver_parameter.py:Moduleparameter.test;out_ifname-cbbe: PASS (74.60 s)
 (02/12) driver_parameter.py:Moduleparameter.test;guids-a200: PASS (74.80 s)
 (03/12) driver_parameter.py:Moduleparameter.test;debug_mask_0-b4e5: PASS (75.02 s)
 (04/12) driver_parameter.py:Moduleparameter.test;debug_mask_1-fe61: PASS (74.56 s)
 (05/12) driver_parameter.py:Moduleparameter.test;debug_mask_2-3b54: PASS (78.15 s)
 (06/12) driver_parameter.py:Moduleparameter.test;debug_mask_3-022d: PASS (76.42 s)
 (07/12) driver_parameter.py:Moduleparameter.test;prof_sel_0-9060: FAIL: ping test failed (73.95 s)
 (08/12) driver_parameter.py:Moduleparameter.test;prof_sel_1-cb4e: FAIL: ping test failed (38.28 s)
 (09/12) driver_parameter.py:Moduleparameter.test;prof_sel_2-8f66: PASS (74.82 s)
 (10/12) driver_parameter.py:Moduleparameter.test;prof_sel_3-3d77: FAIL: ping test failed (74.86 s)
 (11/12) driver_parameter.py:Moduleparameter.test;probe_vf_0-67a8: FAIL: Sysfs check failed  (38.39 s)
 (12/12) driver_parameter.py:Moduleparameter.test;probe_vf_1-712e: FAIL: Sysfs check failed  (75.66 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 5 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 830.01 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-12-03T20.51-ac3b9f4/results.html

**Comment by manvanthar** [comment]
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3917740/job.log)

File: io/driver/driver_parameter.py.data/driver_parameter_mlx4_core.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+netmask:
+peer_ip:
+sysfs_check_required: 1
+value: !mux
+    mlx4_en_only_mode_0:
+       module_param_name: "mlx4_en_only_mode"
+       module_param_value: "0"
+    mlx4_en_only_mode_1:
+       module_param_name: "mlx4_en_only_mode"
+       module_param_value: "1"
```
Comment (narasimhan-v) [inline io/driver/driver_parameter.py.data/driver_parameter_mlx4_core.yaml]: @manvanthar Can you make this similar to https://github.com/avocado-framework-tests/avocado-misc-tests/blob/67ea45171364506869cdce12423559e211aefd07/io/driver/driver_parameter.py.data/driver_parameter_mlx5_ib.yaml

Comment (narasimhan-v) [reply io/driver/driver_parameter.py.data/driver_parameter_mlx4_core.yaml]: And all other parameters in all files as well

**Comment by manvanthar** [comment]
Addressed in PR #1648

**Pull request closed without merging**